sync/class.jetpack-sync-module-users.php

```diff
@@ -177,16 +199,16 @@ function save_user_handler( $user_id, $old_user_data = null ) {
 	}
 
 	function save_user_role_handler( $user_id, $role, $old_roles = null ) {
-		$user = $this->sanitize_user( get_user_by( 'id', $user_id ) );
-
+		$user             = $this->sanitize_user( get_user_by( 'id', $user_id ) );
+		$was_user_created = $this->is_create_user();
 		/**
 		 * Fires when the client needs to sync an updated user
 		 *
 		 * @since 4.2.0
 		 *
 		 * @param object The WP_User object
 		 */
-		do_action( 'jetpack_sync_save_user', $user );
+		do_action( 'jetpack_sync_save_user', $user, $was_user_created );
```

Comment:
The point of this is to prevent the User Saved activity from appearing in the Activity Log along side the User Registered activity when a user is registered. Firing a different sync action would not achieve this, and would instead result in 3 activities when a user is registered instead of 1, namely User Registered, User Saved, and the User Created you are suggesting.